src/main/java/com/example/blog/domain/post/controller/PostController.java

```diff
@@ -0,0 +1,47 @@
+package com.example.blog.domain.post.controller;
+
+import com.example.blog.domain.post.domain.Post;
+import com.example.blog.domain.post.dto.PostCreateReq;
+import com.example.blog.domain.post.service.PostService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/posts")
+@RequiredArgsConstructor
+public class PostController {
+    private final PostService postService;
+
+    @PostMapping
+    public ResponseEntity<Post> createPost(@RequestParam Long userId, @RequestBody PostCreateReq postReq) {
+        Post createdPost = postService.createPost(userId, postReq);
+        return ResponseEntity.ok(createdPost);
+    }
+
+    @GetMapping("/{postId}")
+    public ResponseEntity<Post> getPost(@PathVariable Long postId) {
+        Post post = postService.getPost(postId);
+        return ResponseEntity.ok(post);
+    }
+
+    @PutMapping("/{postId}")
+    public ResponseEntity<Post> updatePost(@PathVariable Long postId, @RequestBody PostCreateReq postReq) {
+        Post updatedPost = postService.updatePost(postId, postReq);
+        return ResponseEntity.ok(updatedPost);
+    }
+
+    @DeleteMapping("/{postId}")
+    public ResponseEntity<Void> deletePost(@PathVariable Long postId) {
+        postService.deletePost(postId);
+        return ResponseEntity.noContent().build();
+    }
+
+    @GetMapping("/search")
+    public ResponseEntity<List<Post>> searchPosts(@RequestParam String keyword) {
```

Comment:
```suggestion
public ResponseEntity<List<Post>> searchPosts(
            @RequestParam(required = false) String title,
            @RequestParam(required = false) String content
            @RequestParam(required = false) String username) {
```
이런식으로 파라미터 받으셔서 세가지 경우를 나누어서 리포지토리에서 검색하시면 어느정도 개선이 될거에요..!

Reply:
저는 Like엔티티를 만들고 likecount의 증감은 comment에서 구현햇습니다! like엔티티는 **누가**, **어떤 게시글의** , **어떤 댓글**에 좋아요를 눌러야하는지 알아야 하므로 외래키를 3개 받아서 구현했어요!